mcp/streamable.go

```diff
@@ -133,7 +133,7 @@ func (h *StreamableHTTPHandler) ServeHTTP(w http.ResponseWriter, req *http.Reque
 			http.Error(w, "Accept must contain 'text/event-stream' for GET requests", http.StatusBadRequest)
 			return
 		}
-	} else if !jsonOK || !streamOK {
+	} else if (!jsonOK || !streamOK) && req.Method != http.MethodDelete {
```

Comment:
Sorry, this is turned into a bit of a mess (not because of this CL, but because of a combination of many CLs adding handing for http method.

Can you add a `// TODO: consolidate with handling of http method below`? Thanks.